unified-runtime/scripts/core/exp-host-task.yml

```diff
@@ -0,0 +1,108 @@
+#
+# Copyright (C) 2024 Intel Corporation
+#
+# Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+# See LICENSE.TXT
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Unified Runtime Experimental API for host tasks"
+ordinal: "100"
+
+--- #--------------------------------------------------------------------------
+type: enum
+extend: true
+typed_etors: true
+desc: "Extension enums to $x_device_info_t to support host tasks."
+name: $x_device_info_t
+etors:
+    - name: ENQUEUE_HOST_TASK_SUPPORT_EXP
+      value: "0x7000"
+      desc: "[$x_bool_t] returns true if the device supports enqueueing host tasks"
+
+--- #--------------------------------------------------------------------------
+type: enum
+extend: true
+desc: "Command Type experimental enumerations."
+name: $x_command_t
+etors:
+    - name: HOST_TASK_EXP
+      value: "0x3000"
+      desc: Event created by $xEnqueueHostTaskExp
+
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Host task flag properties"
+name: $x_exp_host_task_flags_t
+etors:
+    - name: TBD
+      desc: "reserved for future use."
+
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Host task properties"
+name: $x_exp_host_task_properties_t
+base: $x_base_properties_t
+members:
+    - type: $x_exp_host_task_flags_t
+      name: flags
+      desc: "[in] host task flags"
+
+--- #--------------------------------------------------------------------------
+type: enum
+extend: true
+desc: "Structure type experimental enumerations"
+name: $x_structure_type_t
+etors:
+    - name: EXP_HOST_TASK_PROPERTIES
+      desc: $x_exp_host_task_properties_t
+      value: "0x5000"
+
+--- #--------------------------------------------------------------------------
+type: fptr_typedef
+desc: "Host task function."
+name: $x_exp_host_task_function_t
+return: void
+params:
+    - type: void*
+      name: pUserData
+      desc: "[in][out] pointer to data to be passed to callback"
```

Comment:
Would it make sense to also pass in the `hQueue` that's invoking the host task here or is it banned to do anything with that?